Public/Src/Engine/Cache.Plugin.CacheCore/CacheCoreArtifactContentCache.cs

```diff
@@ -369,9 +369,19 @@ public async Task<string> GetDiagnosticsAsync()
             Possible<CasHash, Failure> maybeStored = await maybeOpen.ThenAsync(
                 async cache =>
                 {
+                    Contract.Assert(content.CanSeek);
+                    int addCount = 0;
+
                     var result = await Helpers.RetryOnFailureAsync(
                         async lastAttempt =>
                         {
+                            if (addCount > 0)
+                            {
+                                // Reset stream.
+                                content.Seek(0, SeekOrigin.Begin);
+                            }
+
+                            ++addCount;

```

Comment:
Is this the right ordering for incrementing addCount? If this lambda will only be called after the first failure, then addCount needs to be incremented before the test because we want to re-set the stream on the first re-try? Actually, if this lambed runs only after a failure, you don't need addCount at all and can just always re-set the stream.

Reply:
Never mind, looking at RetryOnFailure, I see that's not how it works. Your lambda gets called even the first time.

---
In reply to: [336236874](https://github.com/microsoft/BuildXL/pull/1062#discussion_r336236874) [](ancestors = 336236874)